File: 30-days-of-code/day8-dictionaries-and-maps/dictionaries-and-maps.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <map>
#include <unordered_map>

int main()
{

    int n;
    std::string phoneNumber, name;
    std::unordered_map<std::string, std::string> phoneBook;

    std::cin >> n;

    for (int i{0}; i < n; i++)
    {
        std::cin >> phoneNumber >> name;
        phoneBook.insert({phoneNumber, name});
    }

    std::string search;
    while (std::cin >> search)
    {
        if (phoneBook.find(search) != std::end(phoneBook))
        {
            std::cout << search << "=" << phoneBook[search] << std::endl;
        }
        else
        {
            std::cout << "Not found" << std::endl;
        }
    }

    return 0;
}

File: 30-days-of-code/day8-dictionaries-and-maps/dictionaries-and-maps.py
#!/bin/python3

from sys import stdin

if __name__ == '__main__':
    phonebook = {}

    n = int(input())

    for i in range(0, n):
        inp = input().split(" ")
        phonebook[inp[0]] = inp[1]

    for search in stdin:
        search = search.rstrip()

        if search in phonebook:
            print(search + "=" + phonebook[search])
        else:
            print("Not found")
